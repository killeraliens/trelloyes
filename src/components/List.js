import React from 'react';
import './List.css';
import Card from './Card';


class List extends React.Component {
  handleClickAddCard = (e) => {
    this.props.onAddCard(this.props.id);
  }

  render() {
    if(this.props.cards) {
      const cardsArr = this.props.cards.map(card => <Card
        key={card.id}
        id={card.id}
        listId={this.props.id}
        title={card.title}
        content={card.content}
        onDeleteCard={this.props.onDeleteCard}
      />)
      return (
        <section id={this.props.id} className='List'>
          <header className='List-header'><h2>{this.props.header}</h2></header>
          <div className='List-cards'>{cardsArr}</div>
          <button type='button' onClick={this.handleClickAddCard}>Add random card</button>
        </section>
      );
    } else {
      return null;
    }
  }
}


export default List;
