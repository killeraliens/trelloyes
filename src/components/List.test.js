import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import List from './List';
import STORE from '../store.js';
import returnCardArr from '../App';

describe('List component', () => {

  it('renders List component', () => {
    const div = document.createElement('div');
    ReactDOM.render(<List />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('does not render List component if cards prop null', () => {
    const tree = renderer.create(<List key='1' id='1' header='a header' cards={null} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders List component UI as expected with cards', () => {
    const tree = renderer.create(<List key='1' id='1' header='a header' cards={[STORE.allCards['a']]} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders List component UI as expected with empty cards array', () => {
    const tree = renderer.create(<List key='1' id='1' header='a header' cards={[]} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

})
