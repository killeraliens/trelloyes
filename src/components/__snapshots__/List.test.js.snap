// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component does not render List component if cards prop null 1`] = `null`;

exports[`List component renders List component UI as expected with cards 1`] = `
<section
  className="List"
  id="1"
>
  <header
    className="List-header"
  >
    <h2>
      a header
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
      id="a"
    >
      <h3>
        First card
      </h3>
      <p>
        lorem ipsum
        <button>
          delete
        </button>
      </p>
    </div>
  </div>
</section>
`;

exports[`List component renders List component UI as expected with empty cards array 1`] = `
<section
  className="List"
  id="1"
>
  <header
    className="List-header"
  >
    <h2>
      a header
    </h2>
  </header>
  <div
    className="List-cards"
  />
</section>
`;
